### ISSUE TITLE
getChildrenPage fails for large folders

### ISSUE THREAD

**Issue by donpwilson** [CloudRail/cloudrail-si-android-sdk#87]
I have upgraded from 2.18.0 to 2.20.0 and have found a problem with getChildrenPage. I suspect it was introduced in 2.19.1. Everything works OK provided there are no large folders. I cannot define "large" precisely but it appears to be folders with more than 100 photos (files).  My function (searchFolders) acts iteratively to search the folder tree and crashes when  getChildrenPage reads one of these large folders.

I show below the errors I get for OneDrive and Google Drive. I have not texted exhaustively but it has worked so far (for folders with 100+ photos) for Dropbox and Box.

I tried with getChildren instead and appeared to get similar errors.

Please let me know if you would like further information.

```09-17 18:58:20.943 11021-14410/com.dpw.photoplayer I/Job_LoadCloudRailData: job-queue-worker-6d9 shouldReRunOnThrowable() run count 1 drive OneDrive
                                                                            com.cloudrail.si.exceptions.AuthenticationException: Bearer access token is empty.
                                                                                at com.cloudrail.si.services.OneDrive.getChildrenPage(OneDrive.java:1367)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.searchFolders(Job_LoadCloudRailData.java:252)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.searchFolders(Job_LoadCloudRailData.java:262)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.searchFolders(Job_LoadCloudRailData.java:262)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.onRun(Job_LoadCloudRailData.java:107)
                                                                                at com.birbit.android.jobqueue.Job.safeRun(Job.java:229)
                                                                                at com.birbit.android.jobqueue.JobHolder.safeRun(JobHolder.java:132)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.handleRunJob(ConsumerManager.java:407)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.access$000(ConsumerManager.java:326)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer$2.handleMessage(ConsumerManager.java:354)
                                                                                at com.birbit.android.jobqueue.messaging.SafeMessageQueue.consume(SafeMessageQueue.java:36)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.run(ConsumerManager.java:389)
                                                                                at java.lang.Thread.run(Thread.java:818)
`
```09-17 18:57:20.211 11021-12654/com.dpw.photoplayer I/Job_LoadCloudRailData: job-queue-worker-7d6 shouldReRunOnThrowable() run count 1 drive Google Drive
                                                                            com.cloudrail.si.exceptions.HttpException
                                                                                at com.cloudrail.si.services.GoogleDrive.getChildrenPage(GoogleDrive.java:1548)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.searchFolders(Job_LoadCloudRailData.java:252)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.searchFolders(Job_LoadCloudRailData.java:262)
                                                                                at com.dpw.photoplayer.jobs.Job_LoadCloudRailData.onRun(Job_LoadCloudRailData.java:107)
                                                                                at com.birbit.android.jobqueue.Job.safeRun(Job.java:229)
                                                                                at com.birbit.android.jobqueue.JobHolder.safeRun(JobHolder.java:132)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.handleRunJob(ConsumerManager.java:407)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.access$000(ConsumerManager.java:326)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer$2.handleMessage(ConsumerManager.java:354)
                                                                                at com.birbit.android.jobqueue.messaging.SafeMessageQueue.consume(SafeMessageQueue.java:36)
                                                                                at com.birbit.android.jobqueue.ConsumerManager$Consumer.run(ConsumerManager.java:389)
                                                                                at java.lang.Thread.run(Thread.java:818)
`

**Comment by PatrickStoklasa** [comment]
Hi Don,

I think we just fixed the issue for GoogleDrive in the version 2.20.1, please try the new version and let me know if the issue still persists.

For OneDrive I suspect we forgot to attach the access token in one of the calls. I will look into this and release a fix as soon as possible.

**Comment by donpwilson** [comment]
I confirm that 2.20.1 fixes the issue with Google Drive

**Comment by donpwilson** [comment]
I was disappointed to find that the issue with OneDrive was not fixed in 2.21.0. When do you expect to release a fix?

**Comment by donpwilson** [comment]
Patrick,

I note that you have recently issued 2.21.3 but that the fix for the OneDrive has still not been included. Is there any chance of a fix soon?

**Comment by PatrickStoklasa** [comment]
Hi Don,

Sorry for the late response! I will make sure a fix will be included in next weeks release.

**Comment by PatrickStoklasa** [comment]
The issue should now be fixed in the latest version (2.21.6). Let me know if you still have trouble.

**Comment by donpwilson** [comment]
I confirm that this is now fixed
